R/data.table.R

```diff
@@ -2335,21 +2335,18 @@ copy = function(x) {
   newx = .Call(Ccopy,x)  # copies at length but R's duplicate() also copies truelength over.
                          # TO DO: inside Ccopy it could reset tl to 0 or length, but no matter as selfrefok detects it
                          # TO DO: revisit duplicate.c in R 3.0.3 and see where it's at
-  if (!is.data.table(x)) {
-    # fix for #1476. TODO: find if a cleaner fix is possible..
-    if (is.list(x)) {
-      anydt = vapply_1b(x, is.data.table, use.names=FALSE)
-      if (sum(anydt)) {
-        newx[anydt] = lapply(newx[anydt], function(x) {
-          .Call(C_unlock, x)
-          setalloccol(x)
-        })
-      }
+
+  reallocate = function(y) {
+    if (is.data.table(y)) {
+      .Call(C_unlock, y)
+      setalloccol(y)
+    } else if (is.list(y)) {
+      y[] = lapply(y, reallocate)
```

Comment:
What happens if you have something like `list(data.table(), list(list(list())))`?

Reply:
eventually it either runs into an `is.data.table` branch or by default just does nothing right?